catalog/spatial/registry/registry-federation-admin-service-impl/src/test/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImplTest.java

```diff
@@ -13,6 +13,1727 @@
+                .create(any(CreateRequest.class));
+
+        federationAdminServiceImpl.init();
+        verify(federationAdminServiceImpl).addRegistryEntry(any(Metacard.class));
+    }
+
+    @Test
+    public void initWithMarshalException() throws Exception {
+        QueryRequest request = getTestQueryRequest();
+        QueryResponse response = getPopulatedTestQueryResponse(request);
+        when(security.getSystemSubject()).thenReturn(subject);
+        when(catalogFramework.query(any(QueryRequest.class))).thenReturn(response);
+        doThrow(ParserException.class).when(parser)
+                .marshal(any(ParserConfigurator.class), any(Object.class), any(OutputStream.class));
+
+        federationAdminServiceImpl.init();
+        verify(federationAdminServiceImpl, never()).addRegistryEntry(any(Metacard.class));
+    }
+
+    @Test
+    public void testAddRegistryEntry() throws Exception {
+        String destination = TEST_DESTINATION;
+        String metacardId = TEST_METACARD_ID;
+        Metacard metacard = getTestMetacard();
+        metacard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.REGISTRY_ID,
+                RegistryObjectMetacardType.REGISTRY_ID));
+
+        metacard.setAttribute(new AttributeImpl(Metacard.TAGS,
+                Collections.singletonList(RegistryConstants.REGISTRY_TAG)));
+
+        Metacard createdMetacard = metacard;
+        createdMetacard.setAttribute(new AttributeImpl(Metacard.ID, metacardId));
+
+        Set<String> destinations = new HashSet<>();
+        destinations.add(destination);
+
+        Subject systemSubject = security.getSystemSubject();
+        Map<String, Serializable> properties = new HashMap<>();
+        properties.put(SecurityConstants.SECURITY_SUBJECT, systemSubject);
+        CreateRequest request = new CreateRequestImpl(Collections.singletonList(metacard),
+                properties,
+                destinations);
+        CreateResponse response = new CreateResponseImpl(request,
+                null,
+                Collections.singletonList(createdMetacard));
+
+        when(catalogFramework.create(any(CreateRequest.class))).thenReturn(response);
+
+        String createdMetacardId = federationAdminServiceImpl.addRegistryEntry(metacard,
+                destinations);
+
+        assertThat(createdMetacardId, is(equalTo(metacardId)));
+        verify(catalogFramework).create(any(CreateRequest.class));
+    }
+
+    @Test
+    public void testAddRegistryEntryWithNullDestinations() throws Exception {
+        String metacardId = TEST_METACARD_ID;
+        Metacard metacard = getTestMetacard();
+        metacard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.REGISTRY_ID,
+                RegistryObjectMetacardType.REGISTRY_ID));
+
+        metacard.setAttribute(new AttributeImpl(Metacard.TAGS,
+                Collections.singletonList(RegistryConstants.REGISTRY_TAG)));
+
+        Metacard createdMetacard = metacard;
```

Comment:
I'm a little confused as to why you create the `createdMetacard` variable on these tests. Is there a reason you cannot keep using `metacard`?